### PR TITLE
ci: Add HNS restart to Windows CNIv1 in Load Test Pipeline 

### PR DESCRIPTION
**Pull request by jpayne3506**
**Reason for Change**:

Adds HNS coverage to windows cniv1 in the load test pipeline. 

**Issue Fixed**:

**Requirements**:

- [x] uses [conventional commit messages](https://www.conventionalcommits.org/)
  
- [ ] includes documentation
- [ ] adds unit tests
- [x] relevant PR labels added

**Notes**:

### CODE REVIEW

File: .pipelines/cni/load-test-templates/restart-hns-template.yaml [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+      addSpnToEnvironment: true
+      inlineScript: |
+        make -C ./hack/aks set-kubeconf AZCLI=az CLUSTER=${{ parameters.clusterName }}
+        kubectl get pod -owide -A
+
+        echo "Validate pod IP assignment before HNS restart"
+        make test-validate-state CNI_TYPE=${{ parameters.cni }} OS_TYPE=windows
+
+        podList=`kubectl get pods -n kube-system -l os=windows -owide | grep privileged | awk '{print $1}'`
+        echo Restart HNS service on each node
```
Comment (paulyufan2) [inline .pipelines/cni/load-test-templates/restart-hns-template.yaml:20]: nit: should we quote this message:
 echo "Restart HNS service on each node"

File: .pipelines/cni/load-test-templates/restart-hns-template.yaml [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+        for pod in $podList; do
+          kubectl exec -i -n kube-system $pod -- powershell Restart-Service hns
+        done
+
+        echo "Validate pod IP assignment after HNS restart"
+        make test-validate-state CNI_TYPE=${{ parameters.cni }} OS_TYPE=windows
```
Comment (paulyufan2) [inline .pipelines/cni/load-test-templates/restart-hns-template.yaml:26]: thinking to check more after HNS is restarted. For example, we can check Get-Service HNS from windows?

File: .pipelines/cni/load-test-templates/restart-hns-template.yaml [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+      scriptType: "bash"
+      addSpnToEnvironment: true
+      inlineScript: |
+        make -C ./hack/aks set-kubeconf AZCLI=az CLUSTER=${{ parameters.clusterName }}
+        kubectl get pod -owide -A
+
+        echo "Validate pod IP assignment before HNS restart"
+        make test-validate-state CNI_TYPE=${{ parameters.cni }} OS_TYPE=windows
+
+        podList=`kubectl get pods -n kube-system -l os=windows -owide | grep privileged | awk '{print $1}'`
```
Comment (vipul-21) [inline .pipelines/cni/load-test-templates/restart-hns-template.yaml:19]: nit: I think you can get the privileged pods using label selector instead of grep.

File: .pipelines/cni/pipeline.yaml [hunk 4]
```diff
@@ -160,7 +160,7 @@ stages:
     condition: always()
     dependsOn:
       - cilium_overlay
-      - win22_cniv1
+      - win22_cniv1_HNS
```
Comment (vipul-21) [inline .pipelines/cni/pipeline.yaml:163]: Why `HNS` to the name ?

Comment (jpayne3506) [reply .pipelines/cni/pipeline.yaml:163]: Adding a conditional for the jobs to be run on the cniv1 template was not working as intended. So, I used a [conditional](https://github.com/Azure/azure-container-networking/blob/74f59240b3c5416793a321f3f72b86e6959d217c/.pipelines/cni/singletenancy/cniv1-template.yaml#L158) for a stage and it worked as intended. This resulted in the last stage being ran for the windows E2E being `win22_cniv1_HNS`. 

The intent was to not have additional jobs dependencies and jobs queued for linux cniv1. Azure pipelines was not agreeing with the additional 6 jobs that were only queued for windows.

File: .pipelines/cni/load-test-templates/restart-hns-template.yaml (hunk 3 above)
Comment (jpayne3506) [reply .pipelines/cni/load-test-templates/restart-hns-template.yaml:19]: You can, but the main purpose of the grep here is to remove the headers, normally done with `--no-headers`.

File: .pipelines/cni/load-test-templates/restart-hns-template.yaml (hunk 2 above)
Comment (jpayne3506) [reply .pipelines/cni/load-test-templates/restart-hns-template.yaml:26]: From my findings the only thing that shows that HNS has been restarted was the[ event log ](https://github.com/Azure/azure-container-networking/blob/4def1fbe1de352f088f5b5c4bcf4bd7a93f8ac0e/.pipelines/cni/load-test-templates/restart-hns-template.yaml#L22C11-L22C133). I agree though that we should have more through checks/tests for HNS. For our purposes though the tests that follow the restart do confirm that HNS is working as intended.

File: .pipelines/cni/load-test-templates/restart-hns-template.yaml [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+      scriptType: "bash"
+      addSpnToEnvironment: true
+      inlineScript: |
+        make -C ./hack/aks set-kubeconf AZCLI=az CLUSTER=${{ parameters.clusterName }}
+        kubectl get pod -owide -A
+
+        echo "Validate pod IP assignment before HNS restart"
+        make test-validate-state CNI_TYPE=${{ parameters.cni }} OS_TYPE=windows
+
+        podList=`kubectl get pods -n kube-system -l os=windows -owide --no-headers | awk '{print $1}'`
```
Comment (vipul-21) [inline .pipelines/cni/load-test-templates/restart-hns-template.yaml:19]: This should include label for privileged pod too right ?

Comment (jpayne3506) [reply .pipelines/cni/load-test-templates/restart-hns-template.yaml:19]: Yes, I didn't realize that our noop deployment also has this same label. Pairing the labels should prevent any edge cases in the future.

**Review by vipul-21 — APPROVED** [review]

**Review by paulyufan2 — APPROVED** [review]
